js/5etools-backgrounds.js

```diff
@@ -78,6 +78,29 @@ function d20plusBackgrounds () {
 
 		return [noteContents, gmNotes];
 	};
+
+	// The popup menu for choosing traits, ideals, bonds and flaws
+	// Needs to be its own thing due to having a choose randomly button
+	d20plus.backgrounds.traitMenu = async function(ptrait, ideal, bond, flaw) {
+		// Arguments to send
+		ptraitargs = {
+			countMin: 0,
+			countMax: 2,
+		}
+		args = {
+			countMin: 0,
+			countMax: 1,
+		}
+		
+		// Call the menu
+		pt = await d20plus.ui.chooseCheckboxList(ptrait, "Personality Trait", ptraitargs);
+		id = await d20plus.ui.chooseCheckboxList(ideal, "Ideal", args);
+		bd = await d20plus.ui.chooseCheckboxList(bond, "Bond", args);
+		fl = await d20plus.ui.chooseCheckboxList(flaw, "Flaw", args);
+
+		//Return
+		return [pt, id, bd, fl];
```

Comment:
prefer returning "named" objects over arrays, i.e. `{personalityTrait: pt, ...}`, which can then be destructured as `const {personalityTrait, ...} = output` for better maintainability/readability